### PR TITLE
Backup

### PR DESCRIPTION
**Pull request by Nachtzuster**

### CODE REVIEW

**Comment by lloydbayley** [comment]
Just tested this. Seemed to work well, however, on restore, the birds.db has the wrong permissions. Should be 666 but restored as 644. Found this by trying to delete a recording after restoring. Gave a read only error.
Manually changed to 666 and things worked again.
I'm pulling an all-nighter to try to fix my sleeping patterns so I won't attempt any difficult code fixes just now! <yawn>

**Comment by lloydbayley** [comment]
Ok. Adding a -p to the restore tar command seems to fix it! Although, with lack of sleep, I may not be able to be trusted!

File: scripts/backup_data.sh [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  for obj in  "${objects[@]}";do
+    part2=$(basename "$obj")
+    ! (echo $arch_list | grep -F -q "$part2") && echo corrupted backup file && exit 1
+  done
+}
+
+restore() {
+  echo "Starting restore"
+  for obj in  "${objects[@]}";do
+    tar --extract -f "$ARCHIVE" -C "$(dirname "$obj")" "$(basename "$obj")"
```
Comment (lloydbayley) [inline scripts/backup_data.sh:63]: I believe it needs a -p here to restore permissions.

**Comment by alexbelgium** [comment]
Looks a really nice feature, thanks !

**Comment by lloydbayley** [comment]
.

File: scripts/backup_data.sh [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+  for obj in  "${objects[@]}";do
+    part2=$(basename "$obj")
+    ! (echo $arch_list | grep -F -q "$part2") && echo corrupted backup file && exit 1
+  done
+}
+
+restore() {
+  log "Starting restore"
+  for obj in  "${objects[@]}";do
+    tar --extract - p -f "$ARCHIVE" -C "$(dirname "$obj")" "$(basename "$obj")"
```
Comment (lloydbayley) [inline scripts/backup_data.sh:79]: There's a space between - and p that shouldn't be there...

Comment (Nachtzuster) [reply scripts/backup_data.sh:79]: doh!

**Comment by alexbelgium** [comment]
Nice feature thanks !

**Comment by Emmo213** [comment]
bzipping would provide smaller backup files.


```
tar -cjf birds.tar.bz2 scripts/birds.db
```

```
-rw-r--r--  1 birdnet birdnet 26808320 Oct 29 09:36 birds.tar
-rw-r--r--  1 birdnet birdnet  2314723 Oct 29 09:37 birds.tar.bz2
```

**Comment by Nachtzuster** [comment]
I did consider compression briefly, however mp3 and png files, the biggest part of an actual backup by a large margin, are not compressible. Even the birds.db database, the only notable file that would be worth compressing, is dwarfed by these.
Using compression would not result in any smaller backup file, but would cost _lots_ of cpu cycles, hence time.

**Pull request merged**